src/Arcus.Templates.Tests.Integration/Fixture/TemplateProject.cs

```diff
@@ -0,0 +1,216 @@
+        {
+            var templateFile = new FileInfo(Path.Combine(templateDir.FullName, ".template.config", "template.json"));
+            if (!templateFile.Exists)
+            {
+                throw new FileNotFoundException(
+                    $"Cannot find a correct template project at: {templateDir.FullName} because no './.template.config/template.json' was found");
+            }
+
+            string json = File.ReadAllText(templateFile.FullName);
+            JObject templateJson = JObject.Parse(json);
+            if (!templateJson.TryGetValue("shortName", out JToken shortNameJson))
+            {
+                throw new JsonException(
+                    "Cannot create a custom project from the template project because no 'shortName' element found in the template.json file");
+            }
+
+            return shortNameJson.Value<string>();
+        }
+
+        /// <summary>
+        /// Run the created project from the template with a given set of <paramref name="commandArguments"/>.
+        /// </summary>
+        /// <param name="buildConfiguration">The build configuration on which the project should be build.</param>
+        /// <param name="commandArguments">The command line arguments that control the startup of the project.</param>
+        protected void Run(string buildConfiguration, string commandArguments)
+        {
+            if (_started)
+            {
+                throw new InvalidOperationException("Test demo project from template is already started");
+            }
+
+            RunDotNet($"build -c {buildConfiguration} {WorkingDirectory.FullName}", "Cannot build created project from template");
+            
+            string runCommand = $"exec {Path.Combine(WorkingDirectory.FullName, $"bin/{buildConfiguration}/netcoreapp2.2/{ProjectName}.dll")} {commandArguments ?? String.Empty}";
+            _outputWriter.WriteLine("> dotnet {0}", runCommand);
+            var processInfo = new ProcessStartInfo("dotnet", runCommand)
+            {
+                UseShellExecute = false,
+                CreateNoWindow = true,
+                WorkingDirectory = WorkingDirectory.FullName,
+            };
+
+            _process.StartInfo = processInfo;
+
+            _started = true;
+            _process.Start();
+        }
+
+        /// <summary>
+        /// Performs application-defined tasks associated with freeing, releasing, or resetting unmanaged resources.
+        /// </summary>
+        public void Dispose()
+        {
+            if (_disposed)
+            {
+                return;
+            }
+
+            _disposed = true;
+
+            PolicyResult[] results = 
+            {
+                Policy.NoOp().ExecuteAndCapture(() => Disposing(true)),
+                RetryAction(StopProject),
+                RetryAction(UninstallTemplate),
+                RetryAction(DeleteProjectDirectory), 
```

Comment:
Idea - Can we make this configurable or so? Reason why I'm asking is that if a test fails we might want to see the project that was generated or am I going too far?

Reply:
It seems that we already have a good logging on the build machine when installing/building/unintalling the template for the tests: https://dev.azure.com/codit/Arcus/_build/results?buildId=38402

Can you explain what you would make configurable?

LIke configure that the project directory is not deleted after the test so we can manually look into it locally?

Reply:
Configurable in a way that the project don't gets deleted so we can manually inspect the folder locally? For example?

Reply:
To configure for example to not delete the project dir so we can manually inspect it locally?

Reply:
Jup, that is correct!

Reply:
Could help; but I you could also place a breakpoint instead and let the test run afterwards to clean it anyway.

Maybe also something for an issue?

Reply:
Sure, feel free to open one to track this.

Breakpoint certainly works but it could also help troubleshooting issues in Azure DevOps in case of flaky tests.

Reply:
Done #45